tools/bamtofastq/2.0.0/bamtofastq__2.0.0.cwl

```diff
@@ -0,0 +1,322 @@
+  InlineJavascriptRequirement:
+    expressionLib:
+      - var boolean_to_int = function(input_object){
+          if (input_object === true){
+            return 1;
+          } else {
+            return 0
+          }
+        }
+      - var get_script_path = function(){
+          /*
+          Abstract script path, can then be referenced in baseCommand attribute too
+          Makes things more readable.
+          */
+          return "scripts/run-bamtofastq.sh";
+        }
+      - var get_eval_line = function(){
+          /*
+          Get the line eval bam2fastq...
+          */
+          return "eval \"bamtofastq\" '\"\$@\"'\n"
+        }
+  InitialWorkDirRequirement:
+    listing:
+      - entryname: $(get_script_path())
+        entry: |
+          #!/usr/bin/env bash
+
+          mkdir -p "$(inputs.output_dir)"
+          $(get_eval_line())
+
+baseCommand: ["bash"]
+
+arguments:
+  # Script path
+  - valueFrom: "$(get_script_path())"
+    position: -1
+
+inputs:
+  first_mate:
+    label: F1
+    type: string?
+    doc: "matched pairs first mates"
+    inputBinding:
+      prefix: F=
+      separate: false
+  second_mate:
+    label: F2
+    type: string?
+    doc: "matched pairs second mates"
+    inputBinding:
+      prefix: F2=
+      separate: false
+  single_end:
+    label: single end
+    type: string?
+    doc: "single end"
+    inputBinding:
+      prefix: S=
+      separate: false
+  unmatched_pairs_first_mate:
+    label: unmatched pairs first mate
+    type: string?
+    doc: "unmatched pairs first mates"
+    inputBinding:
+      prefix: O=
+      separate: false
+  unmatched_pairs_second_mate:
+    label: unmatched pairs mate2
+    type: string?
+    doc: "unmatched pairs second mates"
+    inputBinding:
+      prefix: O2=
+      separate: false
+  collate:
+    label: collate
+    type: int?
+    doc: "collate pairs"
+    inputBinding:
+      prefix: collate=
+      separate: false
+      valueFrom: "$(boolean_to_int(self))"
+  combs:
+    label: combs
+    type: boolean?
+    doc: "print some counts after collation based processing"
+    inputBinding:
+      prefix: combs=
+      separate: false
+      valueFrom: "$(boolean_to_int(self))"
+  filename:
+    label: file name
+    type: File
+    doc: "input filename"
+    streamable: true
+    inputBinding:
+      prefix: filename=
+      separate: false
+  input_format:
+    label: input format
+    type: string?
+    doc: "input format: cram, bam or sam [bam]"
+    inputBinding:
+      prefix: inputformat=
+      separate: false
+  reference:
+    label: reference
+    type: File?
+    doc: "name of reference FastA in case of inputformat=cram"
+    inputBinding:
+      prefix: reference=
+      separate: false
+  ranges:
+    label: ranges
+    type: string?
+    doc: "input ranges (bam and cram input only, default: read complete file)"
+    inputBinding:
+      prefix: ranges=
+      separate: false
+  exclude:
+    label: exclude
+    type: string?
```

Comment:
Is this not a boolean input? Or do you specify the flag?

Reply:
No - needs a flag as `exclude=<[SECONDARY,SUPPLEMENTARY]>`